### PR TITLE
Minor fixes

### LINKED ISSUE #398: Metadata code does not create proper ExtHandlerPackage objects
Unfortunately the metadata code does not create the proper structures to handle VM extensions, resulting in the following error:

    WARNING Agent WALinuxAgent-2.1.6.1 failed with exception: 'ExtHandlerPackage' object has no attribute 'disallow_major_upgrade'

We need to fix metadata.py# get_ext_handler_pkgs to properly create ExtHandlerPackage objects.

### PR DESCRIPTION
**Pull request by hglkrijger**
Verified changes on a dedicated environment. 

- pep8 cleanup
- adjust logging verbosity for metadata certificate methods
- add missing property `disallow_major_upgrade` to `ExtHandlerPackage`
- fixes #398 

/cc @brendandixon @SRIKKANTH

### CODE REVIEW

File: azurelinuxagent/common/protocol/metadata.py [hunk 1]
```diff
@@ -201,22 +200,29 @@ def get_ext_handlers(self, last_etag=None):
...
-                logger.warn("Failed to get version uris: {0}", e)
-                logger.info("Retry getting version uris")
-        set_properties("extensionPackages", ext_handler_pkgs, data)
-        return ext_handler_pkgs
+                logger.warn("Failed to fetch manifest: {0}", e)
+
+        if manifest is None:
+            raise ValueError("Extension manifest is empty")
+
+        set_properties("extensionPackages", pkg_list, manifest)
```
Comment (brendandixon) [inline azurelinuxagent/common/protocol/metadata.py]: @hglkrijger I'm confused. How does this create an instance of ExtHandlerPackage?

Comment (hglkrijger) [inline azurelinuxagent/common/protocol/metadata.py]: The call to `set_properties` instantiates a new list of `ExtHandlerPackage`, and then populates them with the data from `manifest`.

Comment (brendandixon) [inline azurelinuxagent/common/protocol/metadata.py]: @hglkrijger Got it.

**Comment by brendandixon** [comment]
lgtm

**Pull request merged**